Info-Orbs/src/widgets/webDataWidget.cpp

```diff
@@ -0,0 +1,158 @@
+                        m_color[i] = stringToColor(doc[i]["color"].as<String>());
+                    } else {
+                        m_color[i] = m_defaultColor;
+                    }
+                    if (doc[i].containsKey("labelColor")) {
+                        m_labelColor[i] = stringToColor(doc[i]["labelColor"].as<String>());
+                    } else {
+                        m_labelColor[i] = m_color[i];
+                    }
+                    if (doc[i].containsKey("background")) {
+                    m_background[i] = stringToColor(doc[i]["background"].as<String>());
+                    } else {
+                        m_background[i] = m_defaultBackground;
+                    }
+                }
+                
+                m_lastUpdate = millis();
+            } else {
+                // Handle JSON deserialization error
+                Serial.println("deserializeJson() failed");
+            }
+        } else {
+            // Handle HTTP request error
+            Serial.printf("HTTP request failed, error: %s\n", http.errorToString(httpCode).c_str());
+        }
+        http.end();
+    }
+}
+
+int32_t WebDataWidget::stringToColor(String color) {
+    color.toLowerCase();
+    color.replace(" ", "");
+    if (color == "black") {
+        return TFT_BLACK;
+    } else if (color == "navy") {
+        return TFT_NAVY;
+    } else if (color == "darkgreen") {
+        return TFT_DARKGREEN;
+    } else if (color == "darkcyan") {
+        return TFT_DARKCYAN;
+    } else if (color == "maroon") {
+        return TFT_MAROON;
+    } else if (color == "purple") {
+        return TFT_PURPLE;
+    } else if (color == "olive") {
+        return TFT_OLIVE;
+    } else if (color == "lightgrey") {
+        return TFT_LIGHTGREY;
+    } else if (color == "darkgrey") {
+        return TFT_DARKGREY;
+    } else if (color == "blue") {
+        return TFT_BLUE;
+    } else if (color == "green") {
+        return TFT_GREEN;
+    } else if (color == "cyan") {
+        return TFT_CYAN;
+    } else if (color == "red") {
+        return TFT_RED;
+    } else if (color == "magenta") {
+        return TFT_MAGENTA;
+    } else if (color == "yellow") {
+        return TFT_YELLOW;
+    } else if (color == "white") {
+        return TFT_WHITE;
+    } else if (color == "orange") {
+        return TFT_ORANGE;
+    } else if (color == "greenyellow") {
+        return TFT_GREENYELLOW;
+    } else if (color == "pink") {
+        return TFT_PINK;
+    } else if (color == "brown") {
+        return TFT_BROWN;
+    } else if (color == "gold") {
+        return TFT_GOLD;
+    } else if (color == "silver") {
+        return TFT_SILVER;
+    } else if (color == "skyblue") {
+        return TFT_SKYBLUE;
+    } else if (color == "vilolet") {
+        return TFT_VIOLET;
+    } else {
+        Serial.print("Invalid color: ");
+        Serial.println(color);
+        return TFT_BLACK;
+    }
+}
+
+String WebDataWidget::getFingerprint(int8_t display) {
```

Comment:
I think I am liking "fingerprint" more than "stamp" as I did in weatherWidget. Thoughts?